rpc/handler.go

```diff
@@ -571,7 +574,20 @@ func (h *handler) handleCallMsg(ctx *callProc, msg *jsonrpcMessage) *jsonrpcMess
 		if resp.Error != nil {
 			ctx = append(ctx, "err", resp.Error.Message)
 			if resp.Error.Data != nil {
-				ctx = append(ctx, "errdata", resp.Error.Data)
+				// If the log level is debug, log the full error data. Otherwise, truncate it.
+				if h.log.Enabled(context.Background(), log.LvlDebug) {
+					ctx = append(ctx, "errdata", resp.Error.Data)
+				} else {
+					errDataStr := fmt.Sprintf("%v", resp.Error.Data)
+					// Truncate the error data if it is too long. Otherwise, preserve the original data.
+					if len(errDataStr) > truncateLimit {
+						remaining := len(errDataStr) - truncateLimit
+						errDataStr = errDataStr[:truncateLimit] + "...(truncated remaining " + strconv.Itoa(remaining) + " chars)"
+						ctx = append(ctx, "errdata", errDataStr)
```

Comment:
this would eventually convert the err data into string (it's actually an {}interface) but I think that was a more straightforward approach than using `reflect` package.

Reply:
That `interface{}` is quite annoying!

I think it's clearer to use another `fmt.Sprintf` here:

```go
errDataStr = fmt.Sprintf("%s... (truncated remaining %d chars)", errDataStr[:truncateLimit], remaining)
```

Reply:
yup makes sense